power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -373,19 +409,31 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
     }
 
     // get sequence idx map of a certain batch scenario
-    SequenceIdx get_sequence_idx_map(ConstDataset const& update_data, Idx scenario_idx) const {
+    SequenceIdx get_sequence_idx_map(ConstDataset const& update_data, Idx scenario_idx,
+                                     UpdateCompIndependence comp_independence = {}) const {
         auto const process_buffer_span = [](auto const& buffer_span, auto const& get_sequence) {
             auto const it_begin = buffer_span.begin();
             auto const it_end = buffer_span.end();
             return get_sequence(it_begin, it_end);
         };
 
-        auto const get_seq_idx_func = [&state = this->state_, &update_data, scenario_idx,
-                                       &process_buffer_span]<typename CT>() -> std::vector<Idx2D> {
-            auto const get_sequence = [&state](auto const& it_begin, auto const& it_end) {
-                return main_core::get_component_sequence<CT>(state, it_begin, it_end);
-            };
+        auto const get_seq_idx_func = [&state = this->state_, &update_data, scenario_idx, &process_buffer_span,
+                                       &comp_independence]<typename CT>() -> std::vector<Idx2D> {
+            // TODO: (jguo) this function could be encapsulated in UpdateCompIndependence in update.hpp
+            Idx const n_comp_elements = [&comp_independence]() {
+                if (auto const comp_idx =
+                        std::ranges::find_if(comp_independence, [](auto const& comp) { return comp.name == CT::name; });
+                    comp_idx != comp_independence.end()) {
+                    if ((*comp_idx).no_id_col() || (*comp_idx).no_id_row()) {
+                        return (*comp_idx).get_n_elements();
```

Comment:
not necessary but it's OK to use
```suggestion
                    if (comp_idx->no_id_col() || comp_idx->no_id_row()) {
                        return comp_idx->get_n_elements();
```